packages/itwinui-react/src/utils/components/MiddleTextTruncation.tsx

```diff
@@ -46,33 +46,38 @@ export type MiddleTextTruncationProps = {
 export const MiddleTextTruncation = (props: MiddleTextTruncationProps) => {
   const { text, endCharsCount = 6, textRenderer, style, ...rest } = props;
 
-  const [ref, visibleCount] = useOverflow(text);
-
-  const truncatedText = React.useMemo(() => {
-    if (visibleCount < text.length) {
-      return `${text.substring(
-        0,
-        visibleCount - endCharsCount - ELLIPSIS_CHAR.length,
-      )}${ELLIPSIS_CHAR}${text.substring(text.length - endCharsCount)}`;
-    } else {
-      return text;
-    }
-  }, [endCharsCount, text, visibleCount]);
+  const truncatedText = React.useCallback(
```

Comment:
It's worth noting that converting a `useMemo` into a `useCallback` changes the meaning significantly: previously, the nodes were being memoized, whereas now only the function is memoized, and the nodes are recreated on every render (because the function is called on every render).

It might not be a problem in this case, but I wonder if we could avoid it somehow 🤔